### DISCUSSION TITLE
Enforce Latin Hyper Cube sampler to take on samples from the bounds

### DISCUSSION THREAD

**Discussion by izsahara** [IDAES/idaes-pse#586] [Q&A] [answered]
I have a input data set of size `500,000 x 5`, which has been normalized prior to LHS sampling.
When attempting to draw a LHS of 250 samples, the range of the extracted LHS does not match the original data set range.
Is there a way to enforce the sampler to take on samples at the bounds? 

```
extracted.min(0)
Out[3]: array([-0.9922156 , -0.99270723, -1.01592417, -0.99768577, -0.93719903])
original[:, :5].min(0)
Out[4]: array([-0.99998581, -0.99997099, -1.10260857, -0.99998881, -0.9532046 ])
extracted.max(0)
Out[5]: array([0.98092641, 0.99896162, 0.96639979, 0.98974614, 1.03913968])
original[:, :5].max(0)
Out[6]: array([0.99998061, 0.99998158, 1.01371486, 0.99998586, 1.04649708])
```

**Comment by OOAmusat** [comment, marked as answer]
Hi @izsahara,

The method is performing as expected: the LHS algorithm does not place points at the corners of the sample space; it only enforces that all the samples are within the range covered by the original dataset.

If you explicitly require points at the corners of your space, you can take two approaches:
 

1. Use uniform sampling. Uniform sampling (with edges=True) ensures that points are placed at the corner of the design space.
2. Use LHS and then manually add the boundary points: this should be fairly straightforward (with np.concatenate) since the sampling tool returns a numpy array.

Hopefully one of these solutions is be suitable for your case.

**Reply by izsahara** [reply]
Thanks, second point does the job.
